npc/re/instances/WaveMode.txt

```diff
@@ -0,0 +1,807 @@
+//===== rAthena Script =======================================
+//= Wave Mode
+//===== Description: =========================================
+//= [Walkthrough Conversion]
+//= Wave Mode Forest and Sky Instances
+//=   Officially monsters can use their skills
+//	  (NPC_SUMMONSLAVE, at least). NPC_EMOTION and
+//	  NPC_EMOTION_ON have been blocked since it disrupts the
+//	  walk system of the instance on rAthena.
+//===== Changelogs: ==========================================
+//= 1.0 First version. [Capuche]
+//============================================================
+
+// Function simple to move the monster when idle. Can be improved.
+// Move the monster to the next coordinates + end the script on successful ; kill the monster + return on fail
+// callfunc( "F_mobidle", <npc name + idle label>, <size coord>, <defined x array>, <x shift>,  <defined y array>, <defined spot-spot distance array> );
+function	script	F_mobidle	{
+	sleep2 1000;	// stop if no rid
+
+	.@game_id = getattachedrid();
+	.@dist_min = 300;
+	.@size = getarg(1);
+	.@dx = getarg(3);
+	getunitdata .@game_id, .@data;
+
+	for ( .@i = 0; .@i < .@size; .@i++ ) {
+		.@dist_to_spot[.@i] = distance( .@data[UMOB_X], .@data[UMOB_Y], (getelementofarray( getarg(2),.@i ) + .@dx), getelementofarray( getarg(4),.@i ) );
+		if (.@dist_min >= .@dist_to_spot[.@i]) {
+			.@dist_min = .@dist_to_spot[.@i];
+			.@index = .@i;
+		}
+	}
+
+	if (.@dist_min <= 1) {
+		.@index++;
+		if (.@index >= .@size) {
+			mobchat( .@game_id, 0, 0xFF0000, "" + rid2name(.@game_id) + " : Arrival!" );	// ZC_NPC_CHAT
+			unitskilluseid .@game_id,301,1,.@game_id,1,false;	// SA_INSTANTDEATH
+			return;
+		}
+	}
+	else if (.@index < (.@size -1)) {
+		.@total[0] = .@dist_min + getelementofarray( getarg(5), .@index );
+		.@total[1] = .@dist_to_spot[.@index + 1] + getelementofarray( getarg(5), .@index+1 );
+		if (.@total[0] > .@total[1])
+			.@index = .@index + 1;
+	}
+
+	if (.@dist_to_spot[.@index] > 14) {
+		unitkill .@game_id;
+		end;
+	}
+	if (!.@data[UMOB_TARGETID]) {
+		unitwalk .@game_id, (getelementofarray( getarg(2),.@index ) + .@dx), getelementofarray( getarg(4),.@index );
+		sleep2 50;	// for now a delay between unitwalk and mob_setidleevent is needed
+	}
+	mob_setidleevent .@game_id, getarg(0);
+	end;
+}
+
+prontera,146,75,1	script	Zonda Rep#pron	4_F_ZONDAGIRL,{
+	.@party_id = getcharid(1);
+	.@player_name$ = strcharinfo(0);
+	getmapxy .@map_name$,.@x,.@y, UNITTYPE_PC;
+
+	if (.@party_id && getpartyleader(.@party_id,2) == getcharid(0)) {
```

Comment:
Pretty minor, but maybe the recently added use is_party_leader as other scripts? And obviously the new getmapxy constants change. It could even just be used below as getmapxy wave_mode_map$, wave_move_x, wave_move_y; below as the vars won't serve any use here if the player never really warps, but that sounds too picky.